src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2048,13 +2048,38 @@ def process_declarations(self, parent, nodes, arg_list,
             elif isinstance(node, (Fortran2003.Access_Stmt,
                                    Fortran2003.Derived_Type_Def,
                                    Fortran2003.Stmt_Function_Stmt,
-                                   Fortran2003.Use_Stmt,
-                                   Fortran2003.Implicit_Part)):
-                # These node types are handled separately with the exception
-                # of Implicit_Part which we currently silently ignore
-                # (TODO #1254).
+                                   Fortran2003.Use_Stmt)):
+                # These node types are handled separately
                 pass
-
+            elif isinstance(node, Fortran2003.Implicit_Part):
+                for stmt in node.children:
+                    if isinstance(stmt, Fortran2003.Parameter_Stmt):
+                        for parameter_def in stmt.children[1].items:
+                            name, expr = parameter_def.items
+                            symbol = parent.symbol_table.lookup(str(name))
+                            if not isinstance(symbol, DataSymbol):
```

Comment:
Update: I can't see anything in the standard about this. Perhaps we should at least allow for it so that we fail gracefully (i.e. if `lookup` raises a KeyError)?

Reply:
Done (it puts it in a Codeblock again, instead of failing)